### PR TITLE
Prepare callback fixes

### PR DESCRIPTION
**Pull request by mbouron-gpsw**

### CODE REVIEW

File: libnodegl/node_texture.c [hunk 1]
```diff
@@ -470,6 +470,17 @@ static int texture2d_init(struct ngl_node *node)
     s->params.format = get_preferred_format(gpu_ctx, s->format);
     s->supported_image_layouts = s->direct_rendering ? -1 : (1 << NGLI_IMAGE_LAYOUT_DEFAULT);
 
+    /* Check that the underlying media node is not shared across multiple textures */
```
Comment (cboesch-gpsw) [inline libnodegl/node_texture.c:473]: can  you keep the Android explanation about why we don't do this?

**Review by cboesch-gpsw — COMMENTED** [review]
I think it makes sense to have the prepare-on-all-children done automatically at API level and keep the callbacks simple. OK as-is too but it's kind of a lot of self-imposed boilerplate, in particular since we make mandatory to nodes to do it all the time.

File: libnodegl/node_texture.c (hunk 1 above)
Comment (mbouron-gpsw) [reply libnodegl/node_texture.c]: Done.

**Comment by mbouron-gpsw** [comment]
> I think it makes sense to have the prepare-on-all-children done automatically at API level and keep the callbacks simple. OK as-is too but it's kind of a lot of self-imposed boilerplate, in particular since we make mandatory to nodes to do it all the time.

Unfortunately we can't do that because we need a different state for each branch of the graph (see group_prepare() implementation).

**Review by cboesch-gpsw — APPROVED** [review]

**Pull request merged**